src/javaRosa.js

```diff
@@ -560,10 +560,11 @@ define([
         block.refreshMessages = function () {
             // TODO improve this to display each message beside the
             // form and language to which it applies
-            if (options.messagesPath) {
-                var messages = widgets.util.getMessages(mug, options.messagesPath);
-                $messages.empty().append(messages);
-            }
+            _.each(block.languages, function(lang) {
+                var name = 'itext-' + lang + '-' + block.itextType,
+                    messages = widgets.util.getMessages(mug, name);
+                $('[name='+name+']').parent().parent().find('.messages').empty().append(messages);
```

Comment:
Ha. that doesn't actually work. Right now this only works because `block.itextType` isn't as specific as I thought:

https://github.com/dimagi/Vellum/blob/master/src/javaRosa.js#L2024
https://github.com/dimagi/Vellum/blob/master/src/javaRosa.js#L2106
https://github.com/dimagi/Vellum/blob/master/src/javaRosa.js#L2125

Probably need to rework this to add the error on the widget level.